Index: php-json-schema/php/JsonSchema.php
===================================================================
--- php-json-schema.orig/php/JsonSchema.php	2011-06-25 22:59:36.834274993 +0200
+++ php-json-schema/php/JsonSchema.php	2011-06-25 23:00:20.444274995 +0200
@@ -245,7 +245,7 @@
                             if (preg_match($pattern, $value)) {
                                 $isVali = true;
                             } else {
-                                $this->addError(printf('"%s" does not match "%s"', $value, $pattern));
+                                $this->addError(sprintf('<%s> does not match <%s>', $value, $pattern));
                             }
                         } else {
                             $this->addError('regex pattern is undefined.');
@@ -465,7 +465,7 @@
                 $valueProp = get_object_vars($value);
                 $valueKeys = array_keys($valueProp);
                 $schemaKeys = array_keys($schemaProp);
-                $diffKeys = array_diff($valueKeys, $schemaKeys);
+                $diffKeys = array_diff($schemaKeys, $valueKeys);
                 if (!empty($diffKeys)) {
                     foreach ($diffKeys as $key) {
                         // property not defined / not optional
@@ -473,7 +473,7 @@
                                 || !isset($schemaProp[$key]['optional'])
                                 || !$schemaProp[$key]['optional']
                         ) {
-                            $this->addError(sprintf('key: "%s" is not exist,And it\'s not a optional property.', $value));
+                            $this->addError(sprintf('%s does not exist, and it is not an optional property.', $key));
                             break 2;
                         }
                     }
