--- salt/utils/event.py.orig	2023-04-25 21:46:05.344027000 +0200
+++ salt/utils/event.py	2023-04-25 21:49:42.299542000 +0200
@@ -426,6 +426,8 @@
                 try:
                     self.pusher.connect(timeout=timeout)
                     self.cpush = True
+                except salt.ext.tornado.iostream.StreamClosedError as exc:
+                    log.debug("Unable to connect pusher: %s", exc)
                 except Exception as exc:  # pylint: disable=broad-except
                     log.error(
                         "Unable to connect pusher: %s",
