Index: shairport-2.2.2/Makefile.am
===================================================================
--- shairport-2.2.2.orig/Makefile.am	2015-02-20 13:53:39.772641795 +0000
+++ shairport-2.2.2/Makefile.am	2015-02-20 13:53:53.556641663 +0000
@@ -35,7 +35,3 @@
 if USE_DNS_SD
 shairport_sync_SOURCES += mdns_dns_sd.c
 endif
-
-install-exec-hook:
-	[ -f /etc/init.d/shairport-sync ] || cp scripts/shairport-sync /etc/init.d/
-	update-rc.d shairport-sync defaults 90 10
